src/ast/nodes/shared/FunctionNode.ts

```diff
@@ -100,6 +93,12 @@ export default class FunctionNode extends NodeBase {
 		super.parseNode(esTreeNode);
 	}
 
+	reassignPath(path: ObjectPath) {
+		if (path.length === 1 && path[0] === 'prototype') {
+			this.isPrototypeReassigned = true;
+		}
+	}
```

Comment:
To follow-on the comment about the prototype, ideally would it not be the job of `reassignPath` to track all reassignments to fully distinguish SSA-analyzable paths from non-SSA-analyzable paths?

Further, ideally `reassignPath` should be the same here for both objects and functions surely, apart from some core opt-outs like `length` which would throw on reassignment?

Reply:
Yes and no. At the moment, all paths of a function except the `prototype` are treated as "unknown". So tracking reassignments here would just mean an "unknown" gets replaced by another "unknown" so there is no use doing more analysis.

I thought about adding more logic here but I think it only makes sense if there are at least a few more `function` properties that are known to rollup. Until then, a single flag for the prototype seemed like enough logic.